system-variables.md

```diff
@@ -1,129 +1,836 @@
+    ```sql
+    select * from t, t1 where t.a=t1.a
+    ```
+
+### tidb_opt_write_row_id
+
+- Scope: SESSION
+- Default value: 0
+- This variable is used to set whether to allow `insert`, `replace` and `update` statements to operate on the column `_tidb_rowid`. It is not allowed by default. This variable can be used only when importing data with TiDB tools.
+
+### tidb_projection_concurrency
+
+- Scope: SESSION | GLOBAL
+- Default value: 4
+- This variable is used to set the concurrency of the `Projection` operator.
+
+### tidb_query_log_max_len
+
+- Scope: SESSION
+- Default value: 4096 (bytes)
+- The maximum length of the SQL statement output. When the output length of a statement is larger than the `tidb_query-log-max-len` value, the statement is truncated to output.
+
+Usage example:
+
+```sql
+set tidb_query_log_max_len = 20
+```
+
+### tidb_pprof_sql_cpu <span class="version-mark">New in v4.0</span>
+
+- Scope: SESSION
+- Default value: 0
+- This variable is used to control whether to mark the corresponding SQL statement in the profile output to identify and troubleshoot performance issues.
+
+### tidb_record_plan_in_slow_log
+
+- Scope: SESSION
+- Default value: `1`
+- This variable is used to control whether to include the execution plan of slow queries in the slow log.
+
+### tidb_replica_read <span class="version-mark">New in v4.0</span>
+
+- Scope: SESSION
+- Default value: leader
+- This variable is used to control where TiDB reads data. Here are three options:
+    - leader: Read only from leader node
+    - follower: Read only from follower node
+    - leader-and-follower: Read from leader or follower node
+- See [follower reads](/follower-read.md) for additional details.
+
+### tidb_retry_limit
+
+- Scope: SESSION | GLOBAL
+- Default value: 10
+- This variable is used to set the maximum number of the retries. When a transaction encounters retryable errors (such as transaction conflicts, very slow transaction commit, or table schema changes), this transaction is re-executed according to this variable. Note that setting `tidb_retry_limit` to `0` disables the automatic retry.
+
+### tidb_row_format_version
+
+- Scope: GLOBAL
+- Default value: `2`
+- Controls the format version of the newly saved data in the table. In TiDB v4.0, the [new storage row format](https://github.com/pingcap/tidb/blob/master/docs/design/2018-07-19-row-format.md) version `2` is used by default to save new data.
+- If you upgrade from a TiDB version earlier than 4.0.0 to 4.0.0, the format version is not changed, and TiDB continues to use the old format of version `1` to write data to the table, which means that **only newly created clusters use the new data format by default**.
+- Note that modifying this variable does not affect the old data that has been saved, but applies the corresponding version format only to the newly written data after modifying this variable.
```

Comment:
Can we safely set to a lower value in a newer versioned TiDB cluster? @coocood

Reply:
Yes